src/wallets/hardware/LedgerClient.spec.ts

```diff
@@ -0,0 +1,167 @@
+import { DeepPartial } from "ts-essentials";
+import { mock } from "jest-mock-extended";
+import Transport from "@ledgerhq/hw-transport";
+import TransportWebHID from "@ledgerhq/hw-transport-webhid";
+
+interface CreateLedgerClientParams {
+  client?: DeepPartial<TransportWebHID>;
+  transport?: DeepPartial<Transport>;
+}
+
+const createLedgerClient = (params: CreateLedgerClientParams = {}) => {
+  const client = mock<TransportWebHID>(params.client);
+  const transport = mock<Transport>(params.transport);
+
+  jest.mock("@ledgerhq/hw-transport-webhid", () => {
+    return {
+      ...client,
+      create: () => transport,
+    };
+  });
+
+  const LedgerClient = require("./LedgerClient").default;
+
+  return {
+    client: new LedgerClient(),
+    transport,
+  };
+};
+
+afterEach(() => {
+  jest.resetModules();
+});
+
+describe("getVersion", () => {
+  it("returns the current version", async () => {
+    const { client, transport } = createLedgerClient({
+      transport: {
+        // TODO: Examine real response so this mock is reliable.
+        send: jest.fn().mockResolvedValue(Buffer.from([1, 2, 3])),
+      },
+    });
+
+    await client.connect();
+    const result = await client.getVersion();
+
+    expect(transport.send).toHaveBeenCalledWith(128, 6, 0, 0);
+    expect(result).toEqual("1.2.3");
+  });
+});
+
+describe("getPublicKey", () => {
+  it("returns the public key", async () => {
+    const { client, transport } = createLedgerClient({
+      transport: {
+        // TODO: Examine real response so this mock is reliable.
+        send: jest.fn().mockResolvedValue(Buffer.from([1, 2, 3])),
+      },
+    });
+
+    await client.connect();
+    const result = await client.getPublicKey({
+      derivationPath: "44'/397'/0'/0'/1'",
+    });
+
+    expect(transport.send).toHaveBeenCalledWith(
+      128,
+      4,
+      0,
+      87,
+      // TODO: We should really expect a correctly formatted buffer.
+      //  basically we'll be testing that 'parseDerivationPath' works as expected.
+      expect.any(Buffer)
```

Comment:
I was a bit lazy here using `expect.any`. As the TODO suggests, we'll want to be more precise so we can test the `parseDerivationPath` method is working correctly